CMakeCIPresets.json

```diff
@@ -0,0 +1,608 @@
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release",
+                "CMAKE_CONFIGURATION_TYPES": "Release"
+            }
+        },
+        {
+            "name": "enable-debug",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug",
+                "CMAKE_CONFIGURATION_TYPES": "Debug"
+            }
+        },
+        {
+            "name": "enable-cuda",
+            "hidden": true,
+            "cacheVariables": {
+                "GINKGO_BUILD_CUDA": "ON",
+                "CMAKE_CUDA_COMPILER": "nvcc",
+                "CMAKE_CUDA_ARCHITECTURES": "$env{CUDA_ARCH}"
+            }
+        },
+        {
+            "name": "enable-sycl",
+            "hidden": true,
+            "cacheVariables": {
+                "GINKGO_BUILD_SYCL": "ON"
+            }
+        },
+        {
+            "name": "enable-hip",
+            "hidden": true,
+            "cacheVariables": {
+                "GINKGO_BUILD_HIP": "ON"
+            }
+        },
+        {
+            "name": "enable-openmp",
+            "hidden": true,
+            "cacheVariables": {
+                "GINKGO_BUILD_OMP": "ON"
+            }
+        },
+        {
+            "name": "enable-mpi",
+            "hidden": true,
+            "environment": {
+                "MPI_AS_ROOT": "ON"
+            },
+            "cacheVariables": {
+                "GINKGO_BUILD_MPI": "ON"
+            }
+        },
+        {
+            "name": "enable-gcc",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": "g++"
+            }
+        },
+        {
+            "name": "enable-clang",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": "clang++"
+            }
+        },
+        {
+            "name": "enable-intel",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": "icpc"
+            }
+        },
+        {
+            "name": "enable-dpcpp",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": "dpcpp",
+                "CMAKE_CXX_FLAGS": "-ffp-model=precise"
+            }
+        },
+        {
+            "name": "enable-nvhpc",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": "nvc++",
+                "CMAKE_CXX_FLAGS": "--diag_suppress=useless_using_declaration,declared_but_not_referenced",
+                "CMAKE_CUDA_FLAGS": "-diag-suppress=177"
+            }
+        },
+        {
+            "name": "enable-msvc",
+            "hidden": true,
+            "generator": "Visual Studio 17 2022",
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER": "msvc",
+                "CMAKE_CXX_FLAGS_DEBUG": "/MDd /Zi /Ob1 /O1 /Od /RTC1"
+            }
+        },
+        {
+            "name": "ci-cuda-msvc-nompi-release",
```

Comment:
I chose the naming scheme `ci-[device (multiple connected with +)]-[toolchain]-[nompi|mpi]-[build_type][optional configs]`.  Should this be kept? 
Also, in our ci we seem to use `/` as a separator instead of `-`. We could do the same here, with the side effect, that we would create nested folders.

Reply:
I think it was just a convention we used when building our containers. IMO using `-` seems reasonable. Does having nested folders have any advantages ?

Reply:
The only impact (not sure if it's an advantage) is that the folders used by the Gitlab runner share the same nested structure.